### PR TITLE
2.2/user login

### PR DESCRIPTION
**Pull request by rruzicic**
# Changelog
- Add user login using JWT tokens
- Add configuration management using viper
- Send HTTP request from frontend to backend for login and registration

**WARNING:** ~~Not thoroughly tested~~ Not tested ..... at all

### CODE REVIEW

File: flights/backend/config/config.env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+DB_URI="mongodb://mongo"
+JWT_SECRET="someSecret_addThisToConf"
+JWT_EXP_HRS="1"
+
+PORT=":8080"
```
Comment (Vlodson) [inline flights/backend/config/config.env:5]: Newline

File: flights/backend/models/user.go [hunk 2]
```diff
@@ -1,11 +1,15 @@
 package models
 
+const UserRole = "USER"
+const AdminRole = "ADMIN"
```
Comment (Vlodson) [inline flights/backend/models/user.go:4]: Add [enums](https://www.sohamkamani.com/golang/enums/) instead of this

**Review by Vlodson — CHANGES_REQUESTED** [review]
Can't review FE files because I don't know FE so I'll hope it works

File: flights/backend/repos/user_repo.go [hunk 3]
```diff
@@ -10,10 +10,20 @@ import (
 	"gopkg.in/mgo.v2/bson"
 )
 
-func FindUserByObjectId(id bson.ObjectId) {
+func FindUserByObjectId(id string) {
 
 }
 
+func FindUserByEmail(email string) (models.User, error) {
```
[the remark this answers is not in the document]
Comment (rruzicic) [reply flights/backend/repos/user_repo.go:17]: We actually had two methods that did the same thing one was `FindUserByEmail` and the other was `FindUserByMail` so I just deleted one of them

File: flights/backend/services/user_service.go [hunk 4]
```diff
@@ -6,18 +6,16 @@ import (
...
-	return false
+	user.Role = models.UserRole
+	return repos.CreateUser(user)
 }
 
 func GetAllUsers() []models.User {
 	return repos.FindAllUsers()
 }
 
 func FindUserByMail(mail string) (*models.User, error) {
```
Comment (Vlodson) [inline flights/backend/services/user_service.go:17]: Very minor: change name to FindUserByEmail to match the one from the repo

**Review by Vlodson — CHANGES_REQUESTED** [review]

**Pull request merged**